misaligned.cpp

```diff
@@ -1,21 +1,57 @@
+//#include "PrintColorMapTest.h"
+//#include "PrintColorMapTest.h"
+//#include "ColorMap.h"
 #include <iostream>
+#include <vector>
 #include <assert.h>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+struct ColorPair
+    {
+        int number;
+        std::string majorColor;
+        std::string minorColor;
+    };
+
+std::vector<ColorPair> PrintColorMap() {
+    const char* majorColor[] = { "White", "Red", "Black", "Yellow", "Violet" };
+    const char* minorColor[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
     int i = 0, j = 0;
-    for(i = 0; i < 5; i++) {
-        for(j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+    std::vector<ColorPair> colorMap;
+    for (i = 0; i < 5; i++) {
+        for (j = 0; j < 5; j++) {
+            ColorPair cp;
+            cp.number = i * 5 + j;
+            cp.majorColor = majorColor[i];
+            cp.minorColor = minorColor[j];
+            colorMap.push_back(cp);
+            std::cout << cp.number << " | " << cp.majorColor << " | " << cp.minorColor << "\n";
         }
     }
-    return i * j;
+    return colorMap;
 }
 
-int main() {
-    int result = printColorMap();
+void test_PrintColorMap()
+{
+    std::vector<ColorPair> colorMap = PrintColorMap();
+    int result = colorMap.size();
     assert(result == 25);
+
+    assert(colorMap[0].number == 1);
+    assert(colorMap[0].majorColor == "White");
+    assert(colorMap[0].minorColor == "Blue");
+

```

Comment:
Good asserts for the sample rows! Think about asserting for the delimiters as well, if they are aligned in each row.
You may need to separate the formatting from the printing? Or look for an off-the-shelf solution?